### PR TITLE
[tasks] Debugger support

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

> [!NOTE]
> [Documentation](https://datadoghq.atlassian.net/wiki/spaces/ADX/pages/3896574137/VS+Code+-+Debugging+Agent+Tasks+using+inv+debug)

This adds support for a debugger using [debugpy](https://github.com/microsoft/debugpy) (works for VS Code).

1. Added debug task to trigger remote debugger
2. Updated invoke unit tests to trigger the debugger when executing tests

Example usage in VS Code:

1. `inv debug invoke-unit-tests.run`
2. In VS Code, launch the debugger with the configuration "Remote Debug Tasks"
3. The debugger is attached and the tests run ! The debugger is triggered when there is a failing assert

### CODE REVIEW

File: tasks/debug.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    "localRoot": "${workspaceFolder}",
+                    "remoteRoot": "."
+                }
+            ]
+        }
+    ]
+}""")
+    if verbose:
+        print(color_message('success:', Color.GREEN), 'Debug config generated to .vscode/launch.json')
```
Comment (FlorentClarret) [inline tasks/debug.py:64]: I would add this in the `vscode` collection, what do you think?

Comment (CelianR) [reply tasks/debug.py:64]: Correct, I think that I'll do some refactoring in a future PR as in `setup.py` there are already the settings file (not the launch one but the settings file + task using this should be moved to `vscode.py` as well). Is it ok for you ? I'll update also the documentation

**Review by FlorentClarret — APPROVED** [review]

File: tasks/invoke_unit_tests.py [hunk 2]
```diff
@@ -33,21 +37,44 @@ def run(ctx, tests: str = '', flags: str = '-b'):
...
+    old_environ = os.environ.copy()
+
+    try:
+        # Update env
+        for key, value in TEST_ENV.items():
+            if key not in os.environ:
+                os.environ[key] = value
+
+        loader = unittest.TestLoader()
+        suite = loader.discover('.', pattern=pattern)
```
Comment (chouetz) [inline tasks/invoke_unit_tests.py:67]: nice!

**Review by chouetz — APPROVED** [review]

**Pull request merged**